DFE-Digital/Dfe.Academies.External
# 107918 every page with radios needs a clear if user selects yes with data

Resolves following issue:-
https://dfe-gov-uk.visualstudio.com.mcas.ms/Academies-and-Free-Schools-SIP/_workitems/edit/107918?McasTsid=26110

Also, added email validation to optional 'MainContactOtherEmail' to show used validation warning

## Type of change

## Expected behaviour
On the following 12 pages. if the user selects no for instance, then goes into the page and selects yes, the optional data input is not cleared:-
Join Trust Reasons
Pre-opening support grant
School consultation
Current Financial Year
Financial investigations
Land and buildings
Next financial year
previous financial year
pupil numbers
School contacts
Change School Name
Conversion Target Date

## Steps to reproduce issue (if relevant)
1. Go into one of the affected pages, fill in data, with either yes or no dependent on which option shows the optional data inputs
2. Go into one of the affected pages, fill in data, with either yes or no dependent on which option DOESNT shows the optional data inputs
3. On the associated summary page, the optional data IS showing

## Steps to test this PR
1. Go into one of the affected pages, fill in data, with either yes or no dependent on which option shows the optional data inputs
2. Go into one of the affected pages, fill in data, with either yes or no dependent on which option DOESNT shows the optional data inputs
3. On the associated summary page, the optional data should no longer show

## Prerequisites
n/a
